plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -80,6 +89,27 @@ func defaultPredicates() sets.String {
 		),
 		// Fit is determined by the presence of the Host parameter and a string match
 		factory.RegisterFitPredicate("HostName", predicates.PodFitsHost),
+
+		// Fit is determined by whether or not there would be too many AWS EBS volumes attached to the node
+		factory.RegisterFitPredicateFactory(
+			"MaxEBSVolumeCount",
+			func(args factory.PluginFactoryArgs) algorithm.FitPredicate {
+				// According to Amazon, attaching more than 40 volumes can cause boot failures
+				// TODO: allow for generically parameterized scheduler predicates, because this is a bit ugly
+				var maxEBSVols uint = DefaultMaxEBSVolumes
+				if rawMaxEBSVols := os.Getenv("KUBE_MAX_EBS_VOLS"); rawMaxEBSVols != "" {
```

Comment:
We don't know if the OpenShift Operations team (or other organizations' cluster administrators) want or plan to attach a fixed number of volumes to nodes. This is a way to allow the admins to reserve a fixed number of volumes outside of the scope of the pods.

We really don't like this approach for the long term, but it seems like a decent short term compromise. The scheduler predicate configuration mechanism is not generic at the moment, and I'd like to open a discussion with the community about how it should be designed going forward, but I don't want that to hold up this PR if we can avoid it.